docs/daemons.rst

```diff
@@ -0,0 +1,374 @@
+* 2. Forced termination -- only if ``cancellation_timeout`` is set:
+  * `asyncio.CancelledError` is raised (for async daemons only).
+  * ``cancellation_timeout`` is awaited (if set).
+* 3a. Giving up and abandoning -- only if ``cancellation_timeout`` is set:
+  * A `ResourceWarning` is issued for potential OS resource leaks.
+  * The finalizer is removed, and the object is released for potential deletion.
+* 3b. Forever polling -- only if ``cancellation_timeout`` is not set:
+  * The daemon awaiting continues forever, logging from time to time.
+  * The finalizer is not removed and the object remains blocked from deletion.
+
+The ``cancellation_timeout`` is measured from the point when the daemon
+is cancelled (forced termination begins), not from where the termination
+itself begins; i.e., since the moment when the cancellation backoff is over.
+The total termination time is ``cancellation_backoff + cancellation_timeout``.
+
+.. warning::
+
+    When the operator is exiting, it has its own timeout of 5 seconds
+    for all "hung" tasks. This includes the daemons after they are requested
+    to exit gracefully and all timeouts are reached.
+
+    If the daemon termination takes longer than this for any reason,
+    the daemon will be cancelled (by the operator, not by the daemon guard)
+    regardless of the graceful timeout of the daemon. If this does not help,
+    the operator will be waiting for all hung tasks until SIGKILL'ed.
+
+.. warning::
+
+    If the operator is running in a cluster, there can be timeouts set for a pod
+    (``terminationGracePeriodSeconds``, the default is 30 seconds).
+
+    If the daemon termination is longer than this timeout, the daemons will not
+    be finished in full at the operator exit, as the pod will be SIGKILL'ed.
+
+Kopf itself does not set any implicit timeouts for the daemons.
+Either design the daemons to exit as fast as possible, or configure
+``terminationGracePeriodSeconds`` and cancellation timeouts accordingly.
+
+
+Safe sleep
+==========
+
+For synchronous daemons, it is recommended to use ``stopped.wait()``
+instead of ``time.sleep()``: the wait will end when either the time is reached
+(as with the sleep), or immediately when the stopped flag is set:
+
+.. code-block:: python
+
+    import kopf
+
+    @kopf.daemon('zalando.org', 'v1', 'kopfexamples')
+    def monitor_kex(stopped, **kwargs):
+        while not stopped:
+            stopped.wait(10)
+
+For asynchronous handlers, regular ``asyncio.sleep()`` should be sufficient,
+as it is cancellable via `asyncio.CancelledError`. If cancellation is neither
+configured nor desired, ``stopped.wait()`` can be used too (with ``await``):
+
+.. code-block:: python
+
+    import kopf
+
+    @kopf.daemon('zalando.org', 'v1', 'kopfexamples')
+    async def monitor_kex(stopped, **kwargs):
+        while not stopped:
+            await stopped.wait(10)
+
+This way, the daemon will exit as soon as possible when the :kwarg:`stopped``
```

Comment:
extra backtick at the end of :kwarg:`stopped``

Reply:
Fixed.